observability/src/main/java/io/grpc/observability/logging/GcpLogSink.java

```diff
@@ -78,13 +97,17 @@ public void write(GrpcLogRecord logProto) {
     try {
       GrpcLogRecord.EventType event = logProto.getEventType();
       Severity logEntrySeverity = getCloudLoggingLevel(logProto.getLogLevel());
+      MonitoredResource resource = getResource(locationTags);
```

Comment:
Similar to my comment about the tags being constant, the `MonitoredResource` that we build for every `write` seems redundant. Can we compute the resource just once (in `GcpLogSink`) and use it for the rest of the lifecycle of the app?

Reply:
Made changes to compute `MonitoredResource` just once in `GcpLogSink`.